### PR TITLE
[CCUBE-1116][AC] create indicator label for form.slider and slider, update the stories book

### PR DESCRIPTION
**Pull request by alvinchow97**
[CCUBE-1116][AC] create indicator label for form.slider and slider, update the stories book

**Changes**
Description of changes...

- [keep | delete] branch

**Changelog entry**
- Indicate current selection in `Form.Slider` and `Form.RangeSlider`

**Additional information**

- You may refer to this [ticket](url)(https://jira.ship.gov.sg/browse/CCUBE-1116)

### CODE REVIEW

File: src/input-range-slider/types.ts [hunk 1]
```diff
@@ -15,6 +15,9 @@ export interface BaseSliderProps {
     showSliderLabels?: boolean | undefined;
     sliderLabelPrefix?: string | undefined;
     sliderLabelSuffix?: string | undefined;
+    showIndicationLabels?: boolean | undefined;
+    indicationLabelPrefix?: string | undefined;
+    indicationLabelSuffix?: string | undefined;
```
Comment (qroll) [inline src/input-range-slider/types.ts:20]:
```suggestion
    showIndicatorLabel?: boolean | undefined;
    indicatorLabelPrefix?: string | undefined;
    indicatorLabelSuffix?: string | undefined;
```

File: src/input-range-slider/input-range-slider.tsx [hunk 2]
```diff
@@ -132,8 +136,24 @@ export const InputRangeSlider = ({
...
     };
 
+    const formatIndicationLabel = () => {
+        return (
+            <LabelText>
+                Searching within&nbsp;
+                {indicationLabelPrefix}
+                {selection}
+                {indicationLabelSuffix}
+            </LabelText>
```
Comment (qroll) [inline src/input-range-slider/input-range-slider.tsx:146]: this label should be generic so the text `Searching within` should not be hardcoded
```suggestion
            <LabelText>
                {indicationLabelPrefix}
                {selection}
                {indicationLabelSuffix}
            </LabelText>
```

File: src/input-range-slider/input-range-slider.tsx [hunk 3]
```diff
@@ -132,8 +136,24 @@ export const InputRangeSlider = ({
         );
     };
 
+    const formatIndicationLabel = () => {
+        return (
+            <LabelText>
+                Searching within&nbsp;
+                {indicationLabelPrefix}
+                {selection}
```
Comment (qroll) [inline src/input-range-slider/input-range-slider.tsx:144]: this is an array and needs to be formatted for display

ranged values need to be handled as well

e.g.
for single thumb, `search within 2km`
for double thumbs, `search within 2-3km`
for 3 or more thumbs, `search within 2-10km` (if selection contains multiple values, pick the min and max)

File: src/input-range-slider/input-range-slider.tsx [hunk 4]
```diff
@@ -132,8 +136,24 @@ export const InputRangeSlider = ({
...
+                {indicationLabelSuffix}
+            </LabelText>
+        );
+    };
+
     return (
         <Wrapper {...otherProps}>
+            {showIndicationLabels && (
+                <IndicationLabelContainer>
+                    <div>{formatIndicationLabel()}</div>
```
Comment (qroll) [inline src/input-range-slider/input-range-slider.tsx:154]: is this extra div needed?

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/input-range-slider/input-range-slider.styles.ts [hunk 5]
```diff
@@ -29,6 +29,10 @@ export const LabelContainer = styled.div`
     gap: 1rem;
 `;
 
+export const IndicationLabelContainer = styled.div`
+    margin-bottom: 0.25rem;
```
Comment (qroll) [inline src/input-range-slider/input-range-slider.styles.ts:33]: should be 1 rem based on figma

File: src/input-range-slider/input-range-slider.tsx (hunk 4 above)
Comment (alvinchow97) [reply src/input-range-slider/input-range-slider.tsx:154]: Removed, sry left this during the testing

File: src/input-range-slider/input-range-slider.styles.ts (hunk 5 above)
Comment (alvinchow97) [reply src/input-range-slider/input-range-slider.styles.ts:33]: Based on the figma, the distance between the indicator and the label below should be the same, think the track got height affected, and the label also using margin-top:0.25rem;  to achieve the visible 1rem

File: src/input-range-slider/input-range-slider.tsx (hunk 2 above)
Comment (alvinchow97) [reply src/input-range-slider/input-range-slider.tsx:146]: Noted haha, i always thought that prefix indicated value infront of changing value, etc currency, measurement etc. Thanks for remind

File: src/input-range-slider/types.ts (hunk 1 above)
Comment (alvinchow97) [reply src/input-range-slider/types.ts:20]: Modified

File: src/input-range-slider/input-range-slider.tsx (hunk 3 above)
Comment (alvinchow97) [reply src/input-range-slider/input-range-slider.tsx]: Updated. thanks for pointing out this

File: src/input-range-slider/input-range-slider.styles.ts (hunk 5 above)
Comment (alvinchow97) [reply src/input-range-slider/input-range-slider.styles.ts]: Solved, updated both label (bottom mix and max, and indicator margin to 1rem)

**Review by qroll — APPROVED** [review]

**Pull request merged**